src/readers/CMakeLists.txt

```diff
@@ -0,0 +1,220 @@
+  DESCRIPTION "glTF files reader"
+  HAS_CUSTOM_CODE
+)
+
+declare_f3d_reader(
+  NAME MetaImageReader
+  EXTENSIONS mha mhd
+  MIMETYPES application/vnd.mhd
+  VTK_READER vtkMetaImageReader
+  DESCRIPTION "MetaImage files reader"
+)
+
+declare_f3d_reader(
+  NAME NrrdReader
+  EXTENSIONS nrrd nhdr
+  MIMETYPES application/vnd.nrrd
+  VTK_READER vtkNrrdReader
+  DESCRIPTION "Nrrd files reader"
+)
+
+declare_f3d_reader(
+  NAME OBJReader
+  EXTENSIONS obj
+  MIMETYPES model/obj
+  VTK_READER vtkOBJReader
+  VTK_IMPORTER vtkOBJImporter
+  DESCRIPTION "OBJ files reader"
+  HAS_CUSTOM_CODE
+)
+
+declare_f3d_reader(
+  NAME PLYReader
+  EXTENSIONS ply
+  MIMETYPES application/vnd.ply
+  VTK_READER vtkPLYReader
+  DESCRIPTION "PLY files reader"
+)
+
+declare_f3d_reader(
+  NAME PTSReader
+  EXTENSIONS pts
+  MIMETYPES application/vnd.pts
+  VTK_READER vtkPTSReader
+  DESCRIPTION "PTS files reader"
+)
+
+declare_f3d_reader(
+  NAME STLReader
+  EXTENSIONS stl
+  MIMETYPES model/stl
+  VTK_READER vtkSTLReader
+  DESCRIPTION "STL files reader"
+  HAS_CUSTOM_CODE
+)
+
+declare_f3d_reader(
+  NAME TIFFReader
+  EXTENSIONS tiff tif
+  MIMETYPES application/x-tgif
+  VTK_READER vtkTIFFReader
+  DESCRIPTION "TIFF files reader"
+)
+
+declare_f3d_reader(
+  NAME VRMLReader
+  EXTENSIONS wrl vmrl
+  MIMETYPES model/vrml
+  VTK_IMPORTER vtkVRMLImporter
+  DESCRIPTION "VRML files reader"
+)
+
+declare_f3d_reader(
+  NAME VTKLegacyReader
+  EXTENSIONS vtk
+  MIMETYPES application/vnd.vtk
+  VTK_READER vtkPDataSetReader
+  DESCRIPTION "VTK Legacy files reader"
+)
+
+declare_f3d_reader(
+  NAME VTKXMLReader
+  EXTENSIONS vtp vtu vtr vti vts vtm
+  MIMETYPES application/vnd.vtp application/vnd.vtu application/vnd.vtr application/vnd.vti application/vnd.vts application/vnd.vtm
+  VTK_READER vtkXMLGenericDataObjectReader
+  DESCRIPTION "VTK XML files reader"
+)
+
+if(F3D_MODULE_OCCT)
+  declare_f3d_reader(
+    NAME OCCTReader
+    EXTENSIONS stp step igs iges
+    MIMETYPES model/iges application/vnd.step
+    VTK_READER vtkF3DOCCTReader
+    DESCRIPTION "OpenCascade files reader"
+    HAS_CUSTOM_CODE
+  )
+endif()
+
+if(F3D_MODULE_ASSIMP)
+  declare_f3d_reader(
+    NAME AssimpReader
+    EXTENSIONS fbx dae dxf off
+    MIMETYPES application/vnd.fbx application/vnd.dae image/vnd.dxf application/vnd.off
+    VTK_IMPORTER vtkF3DAssimpImporter
+    DESCRIPTION "Assimp files reader"
+  )
+endif()
+
+if(F3D_MODULE_EXODUS)
+  declare_f3d_reader(
+    NAME ExodusIIReader
+    EXTENSIONS exo ex2 e
+    MIMETYPES application/vnd.exodus
+    VTK_READER vtkExodusIIReader
+    DESCRIPTION "ExodusII files reader"
+    HAS_CUSTOM_CODE
+  )
+endif()
+
+# Generate the code of the F3DReaderInstantiator class
+configure_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/F3DReaderInstantiator.h.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/F3DReaderInstantiator.h"
+  @ONLY)
+
+set(F3D_SUPPORTED_MIME_TYPES "${F3D_SUPPORTED_MIME_TYPES}" CACHE STRING "Supported mime-type list (read-only)" FORCE)
```

Comment:
why in cache ?

Reply:
It's useful to expose it to the user in the CMake GUI in my opinion, and we need the variable in installing.cmake

Reply:
It is needed. The variable is created in `readers/CMakeLists.txt`, then read in `cmake/installing.cmake`. Even a `set` with a `PARENT_SCOPE` cannot work.  
If there is another way to create a variable visible from sibling CMake files, please let me know.